viadot/sources/s3.py

```diff
@@ -1,266 +1,165 @@
-import os
-from typing import Any, Dict, List
+from typing import List
 
+import awswrangler as wr
+import boto3
 import pandas as pd
+import s3fs
 
-from ..config import get_source_credentials
-from ..exceptions import CredentialError
-from .base import Source
-from ..utils import add_viadot_metadata_columns
+from viadot.sources.base import Source
 
 
 class S3(Source):
     """
     A class for pulling data from and uploading to S3.
 
-    You can either connect to the lake in general
-    (`lake = S3(); lake.exists("a/b/c.csv")`),
-    or to a particular path (`lake = S3(path="a/b/c.csv"); lake.exists()`)
-
     Args:
-        credentials (Dict[str, Any], optional): A dictionary containing
-            the following credentials: `account_name`, `tenant_id`,
-            `client_id`, and `client_secret`.
-        config_key (str, optional): The key in the viadot config holding
-            relevant credentials.
+        profile_name (str, optional): The name of the AWS profile. Defaults to None.
+        aws_access_key_id (str, optional): AWS access key id. Defaults to None.
+        aws_secret_access_key (str, optional): AWS secret access key. Defaults to None.
```

Comment:
The credential params should be `credentials` and `config_key`.

Reply:
Moved to another PR: AnnaGerlich#1